### PR TITLE
adding the last version of the logger class

### PR DESCRIPTION
**Pull request by kingussopp**

### CODE REVIEW

File: src/main/java/com/sevenmap/App.java [hunk 1]
```diff
@@ -42,6 +45,9 @@
 
     private Item testElement = new Item(new Vector3f(0, 0, -1.0f), new Vector3f(0, 0, 0), new Vector3f(1, 1, 1), mesh);
 
+    public static Logger logger = null;
```
Comment (l3alr0g) [inline src/main/java/com/sevenmap/App.java:48]: Why do you initialize your logger this way ? My guess is you should call the default constructor (or simply create one)

File: src/main/java/com/sevenmap/App.java [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 import org.lwjgl.glfw.GLFW;
 
+import java.lang.System.Logger;
+import java.io.File;
```
Comment (l3alr0g) [inline src/main/java/com/sevenmap/App.java:6]: Unused import

File: src/main/java/com/sevenmap/logger.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        FileHandler fileHandler;
+        try {
+          fileHandler = new FileHandler("C:\\logfile.log", true);
+          logger.addHandler(fileHandler);
+          logger.setLevel(Level.ALL);
+          SimpleFormatter formatter = new SimpleFormatter();
+          fileHandler.setFormatter(formatter);
+          logger.log(Level.WARNING, "Warning message");
+          logger.log(Level.SEVERE, "Severe message");
+          logger.log(Level.INFO, "Info message from Throwable object", new Throwable("Throwable exception"));
```
Comment (l3alr0g) [inline src/main/java/com/sevenmap/logger.java:18]: I plainly don't understand what you meant to do here, why do you generate these test logs inside the main function ? Why doesn't the class have a `log(message)` method of its own ? The logger class is supposed to be an alternative to the classic `System.out.println()` call, and in this case, I don't see how anyone could use it that way.

File: src/test/java/com/sevenmap/ui/maths/Vector3fTest.java [hunk 4]
```diff
@@ -40,7 +40,7 @@ public void divideTest() {
     }
 
     @Test
-    public void normalizeTest() {
+        public void normalizeTest() {
```
Comment (l3alr0g) [inline src/test/java/com/sevenmap/ui/maths/Vector3fTest.java:43]: Minor detail : please fix this indentation.

**Review by l3alr0g — CHANGES_REQUESTED** [review]
I know this probably took you a long time to figure out, but the requirements for this PR to be merged are not met at all. Please take some time to read the comments I wrote, then message me on discord if you need some help understanding what you did wrong (in this order, not the other way around).

**Comment by kingussopp** [comment]
For the null logger initialisation and the useless imports I was trying to figure out how to implement the logger in the main class(app.java). but when I couldn't, I just - forgot to delete those lines(among others). It seems that I didn't fully understand the purpose of the logger. Will I be replacing every system.out.print() if things went right ?

**Comment by l3alr0g** [comment]
Yes, that is the point of a logger, it is meant to replace the vanilla `System.out.println()` method by allowing you to (for instance) store the logs in a file or simply not display them, something `println` can't do.

**Comment by kingussopp** [comment]
So the way I see it, the logger class wouldn't need a main method, I will be just using simple methods that each will describe a certain step or error.

**Comment by l3alr0g** [comment]
That's right, you got the idea. You get to decide whether these methods will be static or not (if not then the logger obviously needs to be instantiated). I've converted this PR to draft, so any changes/commits you make will be added to it : no need to create a new one.
When you're done fixing everything, mark it as ready for review, I should be notified.

**Comment by kingussopp** [comment]
Ok great, thanks for the help.

File: src/test/java/com/sevenmap/ui/maths/Vector3fTest.java (hunk 4 above)
Comment (l3alr0g) [reply src/test/java/com/sevenmap/ui/maths/Vector3fTest.java:43]: still not fixed

File: src/main/java/com/sevenmap/App.java (hunk 1 above)
Comment (l3alr0g) [reply src/main/java/com/sevenmap/App.java:51]: can you please modify this ?

**Review by l3alr0g — CHANGES_REQUESTED** [review]
Not ready to merge

**Review by l3alr0g — APPROVED** [review]
Approved for merging, everything seems fine

**Pull request merged**